webforj-plugins/webforj-install-bbj-plugin/src/main/java/com/webforj/installer/MavenBinaryInstaller.java

```diff
@@ -82,14 +85,31 @@ static void installMaven(String directory) throws IOException {
     }
 
     // fix the execution flags for Mac and Linux
-    log.info("Fixing execution flags for Mac and linux");
-    Files.setPosixFilePermissions(Path.of(directory + "/" + BINDIR + "/bin/mvn.cmd"),
-        PosixFilePermissions.fromString("rwxrwxr-x"));
+    if (SystemUtils.IS_OS_MAC || SystemUtils.IS_OS_LINUX) {
+      log.info("Fixing execution flags for Mac and linux");
+
+      Set<PosixFilePermission> perms = new HashSet<>();
+      // user permission
+      perms.add(PosixFilePermission.OWNER_READ);
+      perms.add(PosixFilePermission.OWNER_WRITE);
+      perms.add(PosixFilePermission.OWNER_EXECUTE);
+      // group permissions
+      perms.add(PosixFilePermission.GROUP_READ);
+      perms.add(PosixFilePermission.GROUP_EXECUTE);
+      // others permissions removed
+      perms.add(PosixFilePermission.OTHERS_READ); // Non-Compliant
+      perms.remove(PosixFilePermission.OTHERS_WRITE); // Compliant
+      perms.add(PosixFilePermission.OTHERS_EXECUTE); // Compliant
 
-    Files.setPosixFilePermissions(Path.of(directory + "/" + BINDIR + "/bin/mvn"),
-        PosixFilePermissions.fromString("rwxrwxr-x"));
+      Path binDirectory = Path.of(directory, BINDIR, "bin");
 
-    Files.delete(Path.of(fileZip));
+      Files.setPosixFilePermissions(binDirectory.resolve("mvn.cmd"), perms);
```

Comment:
If this is intended for Mac and Linux, then why is there a need to modify mvn.cmd?

Reply:
good question.  Why was he doing it in the first place?
In this project, stuff like this happens, i always get lots of noise about changing this or that.  I left it here so Stephan could join the conversation and tell us why he never tested for Windows even though his comment said it was.

Reply:
@kevinhagel Did you test this on windows yourself ?

Reply:
no, I have not.

Reply:
I guess it ends here.  I can't test it on windows. I have no way of doing that.

Reply:
I still need to perform local testing, after which I will guide Matthew to conduct tests on his Windows machine.

Reply:
Ok, I just got it to finally run.  A licensing problem.
And i installed the fat jar, there are no other jars in the plugins/DWCJ/lib directory
I checked out webforj-hello-world and ran mvn clean install on it.  I can attach the log, you can see the maven beinary installer running successfully.

[webforj-install-bbj-plugin.log](https://github.com/webforj/webforj/files/14603337/webforj-install-bbj-plugin.log)

Reply:
stop ... something wrong with the bbj subdirecotry.

Reply:
Ok, I found it.  Our old friend Stupid Windows back slash file path separator.  The fix for the zip slip problem requires path checking, I did not see that some of the paths arrived in windows form, some in unix form, and since the test was String ...  I'll get it in today. I need to clean up other stuff first